### DISCUSSION TITLE
Will Windows App SDK allow for the creation of widgets in Windows 11?

### DISCUSSION THREAD

**Discussion by BreeceW** [microsoft/WindowsAppSDK#979] [Q&A] [answered]
Today we learned about Windows 11, and a key new feature in it, Windows Widgets.
![Windows Widgets](https://user-images.githubusercontent.com/18747724/123333998-74297880-d510-11eb-8e3a-8f85ef27dbec.png)
The widgets appear to correspond to built-in Windows apps, however, I do not believe it was mentioned whether this surface will be opened to all app developers. My question is, will any developer be able to make a widget like the ones in the picture above? If so, how and when? I would be interested in using XAML and/or WinUI 3 to create widgets, though I am also curious if other technologies would be supported, e.g. classic Win32 controls or a web browser.

Considering that this appears to be the spiritual successor to [Live Tiles](https://github.com/microsoft/WindowsAppSDK/discussions/427), I would hope they would be at least as accessible to third-party developers.

**Comment by robpascual** [comment]
I'm looking forward to this feature too.  I'd love to start working on some widgets sooner than later.

**Comment by tfennel** [comment, marked as answer]
I think this is a great question and absolutely a dialog we should have! What would **you** like to see the developer story for Widgets in Windows 11 be? What would work best for you and why? What scenarios do you envision similar/different/above and beyond what you could do with Live Tiles?

**Reply by maxkatz6** [reply]
There are live tiles and notifications xml-markup. There is adaptive cards. There is timeline json markup (I don't remember, but it might use adaptive cards actually). 
And, of course, there are xaml and html. 

So, please, reuse at least anything from this list instead of inventing it again. And adaptive cards looks like the best choice.

**Reply by ktos** [reply]
It would be cool if live tiles could be reused so UWP apps pinning tiles will be just pinning widgets and everything working as it was working before.

**Reply by float34** [reply]
So you did not have a developer story when the feature was announced? How's that?

**Reply by riverar** [reply]
It was never announced as an extensible area. I have my doubts that it will ever be.

**Reply by Marv51** [reply]
iOS Widgets, based on SwiftUI created by an app extension, seem like an excellent technical approach.

**Reply by BreeceW** [reply]
Personally, I’d like to be able to write widgets in C# and XAML so I can reuse code from my (UWP, but it should work with UWP and classic apps) app to make a widget for it. Hopefully Windows.UI.Xaml (inbox XAML) and WinUI XAML would both be usable. It would also be good if the widgets integrated with Windows apps somehow. Right now, the To Do widget that we see in Widgets just links to the website. If the widget is bundled with your app, hopefully the widget would just open that app. Otherwise, it would be important for the widget to be able to launch the right app instead of a web link.

**Comment by mdtauk** [comment]
I've mentioned this in other topics.  At the moment the Windows 11 widgets UI seems to be a WebView2 panel on a thin Acrylic background, and all the widgets are web based with some JSON.

The Widgets panel should become a space that embraces multiple technologies, all based on existing concepts.

### Web Widgets
URI anchors to web content that responds to a chosen size so it can adapt it's content - with support for offline PWA like content and behaviours which can be documented.  Each web widget would be a self contained WebView that can be arranged among other widgets.

### XAML Widgets
As is the case with GameBar Widgets, these would be XAML views/pages that are declared and bundled with downloaded apps.  They could support background tasks if they are a more interactive or action orientated widget, and adapt to system themes.  When WinUI or UWP apps are installed that provide a widget, then these widgets will be added to the list of available widgets.

### XML Payloads and Adaptive Cards
With Windows 11, any app providing Live Tile functionality will find them gone.  Non live Secondary Tiles could be replaced with Start Menu icons, but there is no replacement for Live Tiles.  However, instead of a Live Tile, the system could render these same XML payloads as a Widget.  The same size options are there, for medium, large, etc.

As well as handling the lost Tile functionality, Adaptive Cards are a good interoperable approach and they have support in key apps like Teams.  They could render with XAML or more likely, HTML and web tech - so could provide interactive actions and "wizard" like tasks without having to leave the widgets UI.
